aodn_cloud_optimised/bin/generic_cloud_optimised_creation.py

```diff
@@ -49,131 +27,236 @@
 from aodn_cloud_optimised.lib.s3Tools import s3_ls
 
 
+# ---------- Pydantic schema for validating the config ----------
+class ClusterConfig(BaseModel):
+    mode: Optional[str] = None
```

Comment:
```suggestion
    mode: Optional[Literal['x', 'y']] = None
```

Where 'x' and 'y' are the supported modes